python/tank/descriptor/io_descriptor/downloadable.py

```diff
@@ -45,6 +46,17 @@ def _post_download(self, download_path):
     _DOWNLOAD_TRANSACTION_COMPLETE_FILE = "install_complete"
 
     def download_local(self):
+        with self.download_manager() as temporary_path:
```

Comment:
again, manager seems to imply some sort of singleton or persistent object. How about something like `open_write_location()`?

Reply:
I <3 the suggestion.